### PR TITLE
[DOC] Update release & configuration doc

### PR DESCRIPTION
**Pull request by PHILO-HE**
Also replaced `oap-project/gluten` with `apache/incubator-gluten` in doc.

### CODE REVIEW

**Comment by PHILO-HE** [comment]
@weiting-chen, could you take a review? Thanks!

File: docs/release.md
Comment (weiting-chen) [inline docs/release.md:1]: please also change the archived releases url to https://github.com/apache/incubator-gluten/releases/tag/

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Gluten: Plugin to Double SparkSQL's Performance
+# Gluten: A Middle Layer for Offloading JVM-based SQL Engines' Execution to Native Engines
 [![OpenSSF Best Practices](https://www.bestpractices.dev/projects/8452/badge)](https://www.bestpractices.dev/projects/8452)
 *<b>This project is still under active development now, and doesn't have a stable release. Welcome to evaluate it.</b>*
```
Comment (weiting-chen) [inline README.md:3]: please also help to insert one blank before this line to address the issue of sentence continuity for OpenSSF badge and the next sentence.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Gluten: Plugin to Double SparkSQL's Performance
+# Gluten: A Middle Layer for Offloading JVM-based SQL Engines' Execution to Native Engines
```
Comment (yaooqinn) [inline README.md:1]:
```suggestion
# Apache Gluten(Incubating): A Middle Layer for Offloading JVM-based SQL Engines' Execution to Native Engines
```

File: README.md (hunk 1 above)
Comment (PHILO-HE) [reply README.md:4]: Fixed, thanks!

File: README.md (hunk 2 above)
Comment (PHILO-HE) [reply README.md]: Fixed. Thanks for your suggestion!

**Review by weiting-chen — APPROVED** [review]
Looks good to me.

**Pull request merged**